### DISCUSSION TITLE
Extension deprecation

### DISCUSSION THREAD

**Discussion by tomaciazek** [tomaciazek/vscode-ansible#43] [Announcements]
With a heavy heart I'm deprecating the extension published under my own name. Not much after the initial release of this extension I've committed to work with Red Hat to deliver a plugin with better support and assurances than I could provide myself. The extension published under Red Hat has already matured to the point that it doesn't make sense any more to keep both extensions alive.

# FAQ
## Is the other extension missing any features that this one has?
No. The other extension is a fork of this one. All features implemented since the fork happened have been carried over to the other extension at the time they were released. Extension published by Red Hat actually has some extra features.

## Why should I follow the migration guide?
Although the extensions share the features, and implementations thereof, there are some configuration differences.

## Why is the transition not automatic?
Since the implementation is so close, there was a plan to implement some automated migration by adding Red Hat's version as a dependency and allowing its built-in mechanism to remove mine. In the end, I concluded that this would in many cases be quite a surprise and it would often result in unexpected behavior (due to differences in configuration).

## Why is there another extension?
Red Hat reached out to me shortly after I published my extension. They wanted to incorporate it in their Ansible DevTools initiative and help with development of the extension. But if they did that under my repository (and my VS Code publisher), they would not be able to promote it under that initiative, as that is under Red Hat's name. According to their policy, anything that goes out officially through Red Hat, must have certain quality assurances etc. Some of those assurances require that they fully control the codebase and the release process. At least that's how I understood an architect from Red Hat that I spoke with.

**Comment by charltonstanley** [comment]
Thanks for the details @tomaciazek! 
I must be doing something wrong...but I switched to the redhat ansible extension and now ansible-lint doesn't seem to work, nor do i get syntax highlighting in yaml files.  What would be the best way to troubleshoot or get help with this? 
Thanks for the awesome extension!

**Reply by charltonstanley** [reply]
Ah i see now what happened. When I switched to the new extension, the language mode on the files wasn't properly recognized as `ansible`, instead it was switched to `yaml`. After switching it to `ansible` my problem went away.
Sorry for the bother! Hopefully this helps somebody else out.

**Reply by webknjaz** [reply]
@charltonstanley unfortunately, this is a popular problem/misunderstanding — please raise a discussion upstream if you have improvement suggestions :)

**Reply by mdebord-dlr** [reply]
Syntax highlighting is still broken for me, it doesn't have a different color for the task file definition module name (ex: command, file, template, etc).  I've ensured it was set for Ansible task, not yaml.

**Comment by kjoyce77** [comment]
Well thank you for your time and dedication.  I would hope you were rewarded in some way, but either way I can tell you that your extension has helped my Ansible development considerably.  With it being official it will be a little easier for me to evangelize it around my company now as well, so this is good for Ansible users as a whole.

**Reply by tomaciazek** [reply]
I wasn't compensated directly in any way - that's against Red Hat's policy too. But they promised to donate to any non-profit open source organization on my behalf. I chose [Python Software Foundation](https://www.python.org/psf-landing/) - I figured that since Ansible and its plugins (mostly) are written in Python, this would be most beneficial for Ansible's community. And Python does need some funding and a lot of development ;)

**Reply by prghix** [reply]
:/

**Comment by hugo-paredes** [comment]
Like everyone else already said, thank you for the hard work.

**Comment by briantist** [comment]
Thanks for all your work on in this @tomaciazek ! I'm happy to hear you are working with the dev tools initiative and I hope you'll continue to offer your insights and expertise.

**Reply by tomaciazek** [reply]
I surely will! And I have an idea for the next nice quality-of-life feature to implement ;)

**Reply by briantist** [reply]
would like to hear about it! link to it when you can :)

**Comment by ganeshrn** [comment]
@tomaciazek Thank you for all the work you did. I look forward to continue collaborating with you and working towards enhancing and improving the Ansible developer experience with the Ansible extension.
